src/Shelly.hs

```diff
@@ -1043,6 +1043,21 @@ show_command exe args =
     quote t | T.any isSpace t = surround '\'' t
     quote t | otherwise = t
 
+-- quote one argument
+quoteOne :: Text -> Text
+quoteOne t =
+    surround '\'' $ T.concatMap mapSingleQuote t
+    where
+      mapSingleQuote c | c == '\'' = "'\\''"
+      mapSingleQuote c | otherwise = T.pack [c]
```

Comment:
not a deal-breaker for Shelly, but would it be more efficient to use the `T.replace` function instead of `concatMap` ?

Reply:
Indeed, did a little testing, replace is significantly faster on large (300k+ chars) input.